doc/OnlineDocs/advanced_topics/sos_constraints.rst

```diff
@@ -0,0 +1,300 @@
+Special Ordered Sets (SOS)
+==========================
+
+Pyomo allows users to declare special ordered sets (SOS) within their problems.
+These are sets of variables among which only a certain number of variables can
+be non-zero, and those that are must be adjacent according to a given order. 
+
+Special ordered sets of types 1 (SOS1) and 2 (SOS2) are the classic ones, but 
+the concept can be generalised: a SOS of type N cannot have more than N of its 
+members taking non-zero values, and those that do must be adjacent in the set. 
+These can be useful for modelling and computational performance purposes.
+
+By explicitly declaring these, users can keep their formulations and respective 
+solving times shorter than they would otherwise, since the logical constraints
+that enforce the SOS do not need to be implemented within the model and are 
+instead (ideally) handled algorithmically by the solver.
+
+Special ordered sets can be declared one by one or indexed via other sets.
+
+Non-indexed Special Ordered Sets
+--------------------------------
+
+A single SOS of type **N** involving all members of a pyomo Var component can 
+be declared in one line:
+
+.. currentmodule:: pyomo.environ
+
+.. doctest::
+   :hide:
+
+   >>> import pyomo.environ as pyo
+   >>> model = pyo.AbstractModel()
```

Comment:
I would argue that using `ConcreteModel`s makes it harder to understand what the code is doing, since I have to provide the data in the declaration too. While trying to understand how the SOSConstraint class works, I ultimately relied on `ConcreteModel`s, namely to check which set was actually being used for what, but to provide a high-level explanation to others `AbstractModel`s feel adequate.

Reply:
That works for me.  Please make (at least the first) declaration of the `model` not hidden so that it is clear to users that you are demonstrating with an `AbstractModel`